File: vortexdbg-api/src/main/java/com/vortexdbg/signal/AbstractSignalTask.java
package com.vortexdbg.signal;

import com.vortexdbg.thread.BaseTask;

public abstract class AbstractSignalTask extends BaseTask implements com.vortexdbg.signal.SignalTask {

    protected final int signum;

    public AbstractSignalTask(int signum) {
        this.signum = signum;
    }

    @Override
    protected final String getStatus() {
        return "Signal: " + signum;
    }

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/signal/AbstractSignalTask.kt
package com.vortexdbg.signal

import com.vortexdbg.thread.BaseTask

abstract class AbstractSignalTask(@JvmField protected val signum: Int) : BaseTask(), com.vortexdbg.signal.SignalTask {

    protected final override fun getStatus(): String {
        return "Signal: $signum"
    }

}
